tests/io-reader-skip-up-to-test.toit

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import expect show *
```

Comment:
Here are a couple of places, where we'd like to use this:

https://github.com/toitware/cellular/blob/main/src/modules/ublox/ublox.toit#L334

https://github.com/toitware/cellular/blob/main/src/base/at.toit#L397

In both cases, we want the delimiter to also be skipped. I think including the delimiter (if found) in the result is fine.